### DISCUSSION TITLE
Why not React?

### DISCUSSION THREAD

**Discussion by indicozy** [chen-rn/CUA#52] [Ideas]
Hi there!

I like your idea and I'm looking forward for Tauri support!

Is there any trade-off using NextJS? AFAIK we cannot use SSR with Tauri and React Native. Why not to just use React?

**Comment by indicozy** [comment]
Follow-up: is it possible to use Fastify or other backend framework instead of NextJS API?

**Reply by chen-rn** [reply]
It's surely possible! Though, NextJs makes it a lot easier to use tRPC as our backend/frontend glue.

In moe complex applications(maybe you need a lot of sockets), you may want to do something like that instead of NextJs's backend!

**Comment by chen-rn** [comment]
Hey! There is actually Tauri Support now!

You can build the nextjs app into a Tauri app by running the command [here!](https://github.com/chen-rn/CUA/blob/main/apps/nextjs/package.json#:~:text=%22dev%3Atauri%22%3A%20%22yarn%20tauri%20dev%22%2C)

So, on the desktop side, it _is_ React Native transpiled into React done by React Native Web! So in some sense, we _are_ using React + Tauri :) TLDR. Your <View>s in React Native are smartly transformed into <div>s when building Tauri/Next. I'm actually not too sure if Tauri supports SSR at this moment maybe @albbus-stack can shed some light there.

As for why can't we just write in React instead of React Native? You _can_ do that if all you care about is web and desktop support. But that won't work for _native_ mobile apps! Which defeats the purpose of this template, which is to write once and have a performant native app on all platforms!

There are other solutions like Capacitor that wrap your React code in a Native Webview and is essentially an app in the form of a browser, but those apps are generally less performant!

I know it's a little convoluted but I hope it clears thing up a little!

**Reply by indicozy** [reply]
Thanks for your answer!

Could you please clarify what 's' means? Also, why mobile would not work without NextJS? Is it because Solito doesn't support it? If not, why not just use React Native Web and move backend to other framework? Or can migrate backend codebase to backend framework without removing NextJS?

For me, CUA is meant to create complex solutions by small team and iterate fast, so performance is not really a priority.

Yeah, capacitor is not an option.

P.s. I can help with documentation. I'm not familiar with Tauri and React Native, but I want to contribute to CUA.

**Reply by chen-rn** [reply]
oh, I wrote < V i e w > s and  < D i v >  s but I think it vanished in the markdown haha.

**_Re: Mobile not work without Next_:**
So, there are a few parties in play here. Web frontend, mobile frontend and backend. Part of the confusion here is that web frontend and backend are both "NextJS" technically. Mobile can definitely work without nextjs frontned, but in our example it's using the nextjs backend. It doens't have much to do with Solito. Solito is more or less just a set of functions that would smartly call React Navigation or Nextjs router depending on your environment.

**_Re: use RNW and move backend to another framework_:**
you _can_ definitely do that. The notion of this particular template is to build a fullstack app that works on all platforms in a performant and easy manner. Using nextjs backend instead of express/fastify/dedicated server backend is IMO good _enough_ for most apps, and makes things a lot easier. But for sure there are application in which you want to use a dedicated backend.

**_Re:For me, CUA is meant to create complex solutions by small team and iterate fast, so performance is not really a priority._**
Exactly! Though, I think performance is only not a priority when it's _good_ enough! Most apps aren't looking to beat their competitors by having better performance, but rather different value props. So for most CRUD apps, this is probably good enough haha.

_**re: contribute**_ That'd be great! If you'd like you can try building something with CUA and document things that you found confusing in a tutorial like manner. And we can incorporate some of those things into the official docs(which I think needs a revamp never the less :))

**Reply by albbus-stack** [reply]
I don't think ssr is supported by Tauri but I will dig deeper. @indicozy If you want to contribute to the docs they can be found [here](https://github.com/albbus-stack/cua-docs); just fork it, modify or add .mdx files and open a PR, it would be amazing 😍

**Reply by indicozy** [reply]
Thanks for clarification! I like your project, wait for my PRs:)
